src/gpuarray_buffer_cuda.c

```diff
@@ -758,7 +776,7 @@ static gpudata *cuda_alloc(gpucontext *c, size_t size, void *data, int flags) {
    * to a multiple of FRAG_SIZE.  This also ensures that if we split a
    * block, the next block starts properly aligned for any data type.
    */
-  if (!(ctx->flags & GA_CTX_DISABLE_ALLOCATION_CACHE)) {
+  if (ctx->max_cache_size != 0) {
```

Comment:
The theano PR does this correctly now, since the problem was in pygpu.

I believe there is documentation about what max_cache_size and initial_cache_size mean, and if not, I should add it.  But I wouldn't explicitly document this combination since it follows naturally from the properties of both limits.